studio/components/interfaces/Database/Schemas/SchemaGraph.tsx

```diff
@@ -0,0 +1,341 @@
+const NODE_WIDTH = 320
+const NODE_ROW_HEIGHT = 40
+
+function TableNode({ data, targetPosition, sourcePosition }: NodeProps<TableNodeData>) {
+  // Important styles is a nasty hack to use Handles (required for edges calculations), but do not show them in the UI.
+  // ref: https://github.com/wbkd/react-flow/discussions/2698
+  const hiddenNodeConnector = '!h-px !w-px !min-w-0 !min-h-0 !cursor-grab !border-0 !opacity-0'
+
+  return (
+    <>
+      {data.isForeign ? (
+        <div className="rounded-lg overflow-hidden">
+          <header className="text-[0.5rem] leading-5 font-bold px-2 text-center bg-yellow-900 text-gray-300">
+            {data.name}
+            {targetPosition && (
+              <Handle
+                type="target"
+                id={data.name}
+                position={targetPosition}
+                className={`${hiddenNodeConnector} !left-0`}
+              />
+            )}
+          </header>
+        </div>
+      ) : (
+        <div className={`w-[${NODE_WIDTH / 2}px] rounded-lg overflow-hidden`}>
+          <header className="text-[0.5rem] leading-5 font-bold px-2 text-center bg-brand-900 text-gray-300">
+            {data.name}
+          </header>
+
+          {data.columns.map((column) => (
+            <div
+              className="text-[8px] leading-5 relative flex justify-between odd:bg-scale-300 even:bg-scale-400"
+              key={column.id}
+            >
+              <span
+                className={`${
+                  column.isPrimary ? 'border-l-2 border-l-brand-900 pl-[6px] pr-2' : 'px-2'
+                } text-ellipsis overflow-hidden whitespace-nowrap`}
+              >
+                {column.name}
+              </span>
+              <span className="px-2">{column.format}</span>
+              {targetPosition && (
+                <Handle
+                  type="target"
+                  id={column.id}
+                  position={targetPosition}
+                  className={`${hiddenNodeConnector} !left-0`}
+                />
+              )}
+              {sourcePosition && (
+                <Handle
+                  type="source"
+                  id={column.id}
+                  position={sourcePosition}
+                  className={`${hiddenNodeConnector} !right-0`}
+                />
+              )}
+            </div>
+          ))}
+        </div>
+      )}
+    </>
+  )
+}
+
+const TablesGraph: FC<{ tables: PostgresTable[] }> = ({ tables }) => {
+  const reactFlowInstance = useReactFlow()
+  const nodeTypes = useMemo(
+    () => ({
+      table: TableNode,
+    }),
+    []
+  )
+
+  useEffect(() => {
+    getGraphDataFromTables(tables).then(({ nodes, edges }) => {
+      reactFlowInstance.setNodes(nodes)
+      reactFlowInstance.setEdges(edges)
+      setTimeout(() => reactFlowInstance.fitView({})) // it needs to happen during next event tick
+    })
+  }, [tables])
+
+  return (
+    <>
+      <div style={{ width: '100%', height: '100%' }}>
```

Comment:
can use tailwind where possible
```suggestion
      <div className="w-full h-full">
```